### PR TITLE
render left sidebar billboards ('display ads') asynchronously

### LINKED ISSUE #19577: Render sidebar_left billboards asynchronously
## Is this feature related to a problem? Please describe.
This was originally surfaced as a [bug report](https://github.com/forem/forem/issues/19504).

We have the `sidebar_left` placement for billboards, but it does not support user-targeting yet. Throughout the application, we have a pattern of loading the main elements of a page synchronously, and then fetching user-specific data asynchronously. We recently updated billboards to render asynchronously on the article#show (`post-comments`, `post_sidebar`). 

## Describe the solution you’d like

Add support to sidebar_left billboards to load asynchronously so that this placement can take advantage of targeting options.

## Definition of Done

- [ ] Load the `sidebar_left` billboard asynchronously
- [ ] Ensure that the non-signed in traffic is edge cached even when doing the async loading. In this non-signed in case we still benefit from caching because we will not be targeting user attributed. We should lower the cache expiration time for this async request down to ~15 minutes to benefit from “freshness” here. (The default is 24 hours unless cleared by user activity).
- [ ] The signed in async request for this billboard placement will not be cached.

## Additional Context

Examples of what was updated to accomplish this for the article#show placements:
- https://github.com/forem/forem/pull/19285
- https://github.com/forem/forem/issues/19325 / https://github.com/forem/forem/pull/19326

- Up to now @Ridhwana has taken on most of the work, so although it may go faster if she continues to take this one, we would like to expand the experience on the team with this part of the app with others. @Ridhwana is up for pairing on this!

### PR DESCRIPTION
**Pull request by dukegreene**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Given all the permutations of how a given billboard (with or without a given org affiliation) might be targeted to a given user, and how those possibilities affect the complexity of managing cache concerns, we would like to be able to serve more of our billboards asynchronously.

This change updates how the `sidebar_left` and `sidebar_left_2` ads are rendered. It also removes the tests for the presence of those ads at the time of the server's response to the stories index request, since the ads haven't showed up at that time.

## Related Tickets & Documents

- Related Issue #19577
- Closes #19577

## QA Instructions, Screenshots, Recordings

If display ads do not show up in the sidebar in your Uffizzi instance:
- log in as admin
- visit /admin and navigate to Customization -> Display Ads
- create ads that are both Published: true and Approved: true for placements Sidebar Left and Sidebar Left 2
- visit home feed again to view the ads, and check for padding above and between them
- *known issue* is that the dropdown in a very short display ad with no image is larger than the ad that contains it. Have not addressed that yet, may deserve its own style bug issue

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [X] I need help with writing tests (eager for team discussion on best e2e approach for billboard testing)

## [optional] What gif best describes this PR or how it makes you feel?

Tabletop RPG gamer saying "you have to respect the sidebar"

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -212,9 +212,9 @@
       # temporary keeping both routes while transitioning (renaming) display_ads => billboards
       get "/display_ads/:placement_area", to: "billboards#show"
     end
-    get "/billboards/:placement_area", to: "billboards#show"
+    get "/billboards/:placement_area", to: "billboards#show", as: :billboard
     # temporary keeping both routes while transitioning (renaming) display_ads => billboards
-    get "/display_ads/:placement_area", to: "billboards#show"
+    get "/display_ads/:placement_area", to: "billboards#show", as: :display_ad
```
Comment (Ridhwana) [inline config/routes.rb:217]: I'm wondering if we need the "as" here for the 'display_ad_route' since we don't have a display_ad_path in the app?

Comment (dukegreene) [reply config/routes.rb:217]: that’s a good point, no need to add two new path methods when only making use of one. will edit in a minute

**Review by Ridhwana — APPROVED** [review]
🚀

**Comment by dukegreene** [comment]
Planning a fast follow to this with the changes in the screenshot, to add specs confirming the correct fetch call will happen when the display ad container element is on the page. Also working importing a more specific version of observeDisplayAds from an initializer rather than calling it globally (not in screenshot: need to add the function export line to the end of the intializer)
Screenshot 2023-07-21 at 4 52 54 PM

**Comment by klardotsh** [comment]
There's some frontend PRs I'm comfortable reviewing based on either my distant-past-by-now background as a frontend dev, or based on overlap with bits of the stack I actively know more about. This is neither of those things; I'm untagging myself because any review I give here would be dreadfully unqualified.

**Review by filleduchaos — APPROVED** [review]
Tested and works great 🎉  We do have a pretty clean system for rendering these asynchronously going at the moment, shout out to Ridhwana

**Comment by dukegreene** [comment]
Seconding PJ's shoutout, @Ridhwana set us up for success here with code that's straightforward to read and reuse

**Pull request merged**